include/ccf/pal/attestation.h

```diff
@@ -442,6 +442,19 @@ namespace ccf::pal
     measurement = claim_measurement.value();
     report_data = custom_claim_report_data.value();
   }
-
 #endif
+
+  class AttestationCollateralFetchingTimeout : public std::exception
+  {
+  private:
+    std::string msg;
+
+  public:
+    AttestationCollateralFetchingTimeout(const std::string& msg_) : msg(msg_) {}
+
+    virtual const char* what() const throw()
```

Comment:
```suggestion
    virtual const char* what() const noexcept
```

`throw()` is deprecated since C++17 and removed in C++20

Reply:
Will fix every occurrence in a separate PR, this is consistent with existing usage.